### PR TITLE
Sc 848 task copy fee adaption from topos to dev

### PR DESCRIPTION
**Pull request by FHieser**
Abstracts the collection of the projectfee into its own function
Directly copied from the topos branch #667

### CODE REVIEW

File: test/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.t.sol [hunk 1]
```diff
@@ -1212,6 +1212,28 @@ contract BondingCurveBaseV1Test is ModuleTest {
...
+            └── Then the _workflowFeeAmount will be added to the projectCollateralFeeCollected
+    */
+    function testInternalProjectFeeCollected_works(uint _workflowFeeAmount)
+        public
+    {
+        uint currentFeeCollected =
+            bondingCurveFundingManager.projectCollateralFeeCollected();
+
+        // Execute Tx
+        bondingCurveFundingManager.exposed_projectFeeCollected(
```
Comment (Zitzak) [inline test/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.t.sol:1230]: Could you please add the emit tests, as well as add it to the gherkin

**Review by Zitzak — CHANGES_REQUESTED** [review]

**Comment by FHieser** [comment]
@Zitzak could you pls rereview

**Review by Zitzak — APPROVED** [review]

**Comment by marvinkruse** [comment]
Just saw this was merged without me being a reviewer, three things:

(1) Why is the downstream contract not version upgraded?
If `BondingCurveBase_v1` changes, the version of all contracts inheriting it also needs to change, i.e. `FM_BC_Bancor_Redeeming_VirtualSupply_v1` and `FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1`, as the code of these also changed subsequently, speaking from a contract standpoint and it not the same anymore, so it needs changing.

(2) The `_projectFeeCollected` function Natspec should have `@notice`, not just `@dev`

(3) The name of the merged commit is not as we agreed.

Please mark me as a reviewer for **any** PRs that go into **dev** until we are all up to speed on using the Inverter Standard, otherwise it's just double work. (@FHieser @Zitzak)

**Comment by marvinkruse** [comment]
This merge-to-dev-commit should be reverted, or did you guys already rebase too many other PRs?

**Comment by Zitzak** [comment]
I did merge it into the feature branch for the External price funding manager. However if you would like to change it than please go ahead and I force push it to the feature branch @marvinkruse

**Pull request merged**